File: 백준/Silver/2012. 등수 매기기/등수 매기기.py
import sys
input = sys.stdin.readline

# 입력 받기
n = int(input())
expected_ranks = [int(input()) for _ in range(n)]

# 기대 등수를 정렬
expected_ranks.sort()

# 실제 등수를 정렬된 순서대로 배정하며 차이를 계산
total_difference = 0
for actual_rank, expected_rank in enumerate(expected_ranks, start=1):
    total_difference += abs(expected_rank - actual_rank)

# 결과 출력
print(total_difference)

File: 백준/Silver/2012. 등수 매기기/등수 매기기.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int n = Integer.parseInt(br.readLine().trim());
        int[] expectedRanks = new int[n];
        
        for (int i = 0; i < n; i++) {
            expectedRanks[i] = Integer.parseInt(br.readLine().trim());
        }

        Arrays.sort(expectedRanks);
        long totalDifference = 0;
        for (int i = 0; i < n; i++) {
            int actualRank = i + 1;
            totalDifference += Math.abs(expectedRanks[i] - actualRank);
        }
        System.out.println(totalDifference);
        
        br.close();
    }
}